packages/ember-glimmer/lib/syntax/render.js

```diff
@@ -81,7 +83,18 @@ class AbstractRenderManager {
 class SingletonRenderManager extends AbstractRenderManager {
   create(definition, args, dynamicScope) {
     let { name, env } = definition;
-    let controller = env.owner.lookup(`controller:${name}`);
+    let controller = env.owner.lookup(`controller:${name}`) || generateControllerFactory(env.owner, name);
+
+    if (dynamicScope.outletState) {
+      let orphan = dynamicScope.outletState.get('__ember_orphans__');
```

Comment:
Pretty sure this needs to be `dynamicScope.rootOutletState` to work correctly, since only the top-level has `__ember_orphans__`. Also, instead of using the generic code and making an intermediate reference, I think I would rather add a specialized `getOrphan` to the root outlet state reference (and only to the root).